### PR TITLE
Migrate pkg/kubelet/pluginmanager to structured logging

### PR DESCRIPTION
**Pull request by qingwave**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Migrate  `pkg/kubelet/pluginmanager`  to structured logging
#### Which issue(s) this PR fixes:
Ref #98976

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/kubelet/pluginmanager/pluginwatcher/example_handler.go [hunk 1]
```diff
@@ -121,7 +121,7 @@ func (p *exampleHandler) EventChan(pluginName string) chan examplePluginEvent {
 }
 
 func (p *exampleHandler) SendEvent(pluginName string, event examplePluginEvent) {
-	klog.V(2).Infof("Sending %v for plugin %s over chan %v", event, pluginName, p.eventChans[pluginName])
+	klog.V(2).InfoS("Sending event for plugin", "pluginName", pluginName, "event", event, "oldEvent", p.eventChans[pluginName])
```
Comment (ehashman) [inline pkg/kubelet/pluginmanager/pluginwatcher/example_handler.go:124]:
```suggestion
	klog.V(2).InfoS("Sending event for plugin", "pluginName", pluginName, "event", event, "channel", p.eventChans[pluginName])
```

This is a channel, not an event I think. We're doing a lookup by plugin in an event channel list.

File: pkg/kubelet/pluginmanager/pluginwatcher/plugin_watcher.go [hunk 2]
```diff
@@ -65,7 +65,7 @@ func (w *Watcher) Start(stopCh <-chan struct{}) error {
 
 	// Traverse plugin dir and add filesystem watchers before starting the plugin processing goroutine.
 	if err := w.traversePluginDir(w.path); err != nil {
-		klog.Errorf("failed to traverse plugin socket path %q, err: %v", w.path, err)
+		klog.ErrorS(err, "Failed to traverse plugin socket path", "socket path", w.path)
```
Comment (ehashman) [inline pkg/kubelet/pluginmanager/pluginwatcher/plugin_watcher.go:68]:
```suggestion
		klog.ErrorS(err, "Failed to traverse plugin socket path", "socketPath", w.path)
```

Should either be `path` or `socketPath`.

/cc @serathius 
wdyt for consistency?

File: pkg/kubelet/pluginmanager/pluginwatcher/plugin_watcher.go [hunk 3]
```diff
@@ -143,10 +143,10 @@ func (w *Watcher) traversePluginDir(dir string) error {
 			}
 			//TODO: Handle errors by taking corrective measures
 			if err := w.handleCreateEvent(event); err != nil {
-				klog.Errorf("error %v when handling create event: %s", err, event)
+				klog.ErrorS(err, "Failed tocreate event", "event", event)
```
Comment (ehashman) [inline pkg/kubelet/pluginmanager/pluginwatcher/plugin_watcher.go:146]:
```suggestion
				klog.ErrorS(err, "error when handling create event", "event", event)
```

File: pkg/kubelet/pluginmanager/pluginwatcher/plugin_watcher.go [hunk 4]
```diff
@@ -171,7 +171,7 @@ func (w *Watcher) handleCreateEvent(event fsnotify.Event) error {
 	}
 
 	if strings.HasPrefix(fi.Name(), ".") {
-		klog.V(5).Infof("Ignoring file (starts with '.'): %s", fi.Name())
+		klog.V(5).InfoS("Ignoring file (starts with '.')", "fileName", fi.Name())
```
Comment (ehashman) [inline pkg/kubelet/pluginmanager/pluginwatcher/plugin_watcher.go:174]: Should this be filename or path?
cc @serathius

File: pkg/kubelet/pluginmanager/pluginwatcher/plugin_watcher.go [hunk 5]
```diff
@@ -181,7 +181,7 @@ func (w *Watcher) handleCreateEvent(event fsnotify.Event) error {
 			return fmt.Errorf("failed to determine if file: %s is a unix domain socket: %v", event.Name, err)
 		}
 		if !isSocket {
-			klog.V(5).Infof("Ignoring non socket file %s", fi.Name())
+			klog.V(5).InfoS("Ignoring non socket file", "fileName", fi.Name())
```
Comment (ehashman) [inline pkg/kubelet/pluginmanager/pluginwatcher/plugin_watcher.go:184]: Same question re: filename or path

File: pkg/kubelet/pluginmanager/reconciler/reconciler.go [hunk 6]
```diff
@@ -122,42 +122,42 @@ func (rc *reconciler) reconcile() {
 			// with the same socket path but different timestamp.
 			for _, dswPlugin := range rc.desiredStateOfWorld.GetPluginsToRegister() {
 				if dswPlugin.SocketPath == registeredPlugin.SocketPath && dswPlugin.Timestamp != registeredPlugin.Timestamp {
-					klog.V(5).Infof(registeredPlugin.GenerateMsgDetailed("An updated version of plugin has been found, unregistering the plugin first before reregistering", ""))
+					klog.V(5).InfoS(registeredPlugin.GenerateMsgDetailed("An updated version of plugin has been found, unregistering the plugin first before reregistering", ""))
```
Comment (ehashman) [inline pkg/kubelet/pluginmanager/reconciler/reconciler.go:125]: Where is `GenerateMsgDetailed` defined? Do we want to just rip it out?

File: pkg/kubelet/pluginmanager/pluginwatcher/plugin_watcher.go (hunk 2 above)
Comment (serathius) [reply pkg/kubelet/pluginmanager/pluginwatcher/plugin_watcher.go:68]: If it is a normal filesystem path (for example UNIX socket path) then we should use `path`.

File: pkg/kubelet/pluginmanager/reconciler/reconciler.go (hunk 6 above)
Comment (qingwave) [reply pkg/kubelet/pluginmanager/reconciler/reconciler.go:125]: `GenerateMsgDetailed` will format the plugins register message, I',m not sure that's ok for `klog.InfoS`
```
func (plugin *PluginInfo) GenerateMsgDetailed(prefixMsg, suffixMsg string) (detailedMsg string) {
	detailedStr := fmt.Sprintf("(plugin details: %v)", plugin)
	return generatePluginMsgDetailed(prefixMsg, suffixMsg, plugin.SocketPath, detailedStr)
}
```

Comment (serathius) [reply pkg/kubelet/pluginmanager/reconciler/reconciler.go:125]: I think we should replace it with
```suggestion
					klog.V(5).InfoS("An updated version of plugin has been found, unregistering the plugin first before reregistering", "plugin", registeredPlugin)
```

File: pkg/kubelet/pluginmanager/reconciler/reconciler.go [hunk 7]
```diff
@@ -122,42 +122,42 @@ func (rc *reconciler) reconcile() {
...
 					unregisterPlugin = true
 					break
 				}
 			}
 		}
 
 		if unregisterPlugin {
-			klog.V(5).Infof(registeredPlugin.GenerateMsgDetailed("Starting operationExecutor.UnregisterPlugin", ""))
+			klog.V(5).InfoS(registeredPlugin.GenerateMsgDetailed("Starting operationExecutor.UnregisterPlugin", ""))
```
Comment (serathius) [inline pkg/kubelet/pluginmanager/reconciler/reconciler.go:133]:
```suggestion
			klog.V(5).InfoS(r"Starting operationExecutor.UnregisterPlugin", "plugin", registeredPlugin)
```

File: pkg/kubelet/pluginmanager/reconciler/reconciler.go [hunk 8]
```diff
@@ -122,42 +122,42 @@ func (rc *reconciler) reconcile() {
...
 			err := rc.operationExecutor.UnregisterPlugin(registeredPlugin, rc.actualStateOfWorld)
 			if err != nil &&
 				!goroutinemap.IsAlreadyExists(err) &&
 				!exponentialbackoff.IsExponentialBackoff(err) {
 				// Ignore goroutinemap.IsAlreadyExists and exponentialbackoff.IsExponentialBackoff errors, they are expected.
 				// Log all other errors.
-				klog.Errorf(registeredPlugin.GenerateErrorDetailed("operationExecutor.UnregisterPlugin failed", err).Error())
+				klog.ErrorS(registeredPlugin.GenerateErrorDetailed("operationExecutor.UnregisterPlugin failed", err), "UnregisterPlugin failed")
```
Comment (serathius) [inline pkg/kubelet/pluginmanager/reconciler/reconciler.go:140]:
```suggestion
				klog.ErrorS(err, "operationExecutor.UnregisterPlugin failed", "plugin", registeredPlugin)
```

File: pkg/kubelet/pluginmanager/reconciler/reconciler.go [hunk 9]
```diff
@@ -122,42 +122,42 @@ func (rc *reconciler) reconcile() {
...
 			}
 			if err == nil {
-				klog.V(1).Infof(registeredPlugin.GenerateMsgDetailed("operationExecutor.UnregisterPlugin started", ""))
+				klog.V(1).InfoS(registeredPlugin.GenerateMsgDetailed("operationExecutor.UnregisterPlugin started", ""))
```
Comment (serathius) [inline pkg/kubelet/pluginmanager/reconciler/reconciler.go:143]:
```suggestion
				klog.V(1).InfoS("operationExecutor.UnregisterPlugin started", "plugin", registeredPlugin)
```

File: pkg/kubelet/pluginmanager/reconciler/reconciler.go [hunk 10]
```diff
@@ -122,42 +122,42 @@ func (rc *reconciler) reconcile() {
...
 			}
 		}
 	}
 
 	// Ensure plugins that should be registered are registered
 	for _, pluginToRegister := range rc.desiredStateOfWorld.GetPluginsToRegister() {
 		if !rc.actualStateOfWorld.PluginExistsWithCorrectTimestamp(pluginToRegister) {
-			klog.V(5).Infof(pluginToRegister.GenerateMsgDetailed("Starting operationExecutor.RegisterPlugin", ""))
+			klog.V(5).InfoS(pluginToRegister.GenerateMsgDetailed("Starting operationExecutor.RegisterPlugin", ""))
```
Comment (serathius) [inline pkg/kubelet/pluginmanager/reconciler/reconciler.go:151]:
```suggestion
			klog.V(5).InfoS("Starting operationExecutor.RegisterPlugin", "plugin", pluginToRegister)
```

File: pkg/kubelet/pluginmanager/reconciler/reconciler.go [hunk 11]
```diff
@@ -122,42 +122,42 @@ func (rc *reconciler) reconcile() {
...
 			err := rc.operationExecutor.RegisterPlugin(pluginToRegister.SocketPath, pluginToRegister.Timestamp, rc.getHandlers(), rc.actualStateOfWorld)
 			if err != nil &&
 				!goroutinemap.IsAlreadyExists(err) &&
 				!exponentialbackoff.IsExponentialBackoff(err) {
 				// Ignore goroutinemap.IsAlreadyExists and exponentialbackoff.IsExponentialBackoff errors, they are expected.
-				klog.Errorf(pluginToRegister.GenerateErrorDetailed("operationExecutor.RegisterPlugin failed", err).Error())
+				klog.ErrorS(pluginToRegister.GenerateErrorDetailed("operationExecutor.RegisterPlugin failed", err), "RegisterPlugin failed")
```
Comment (serathius) [inline pkg/kubelet/pluginmanager/reconciler/reconciler.go:157]:
```suggestion
				klog.ErrorS(err, "operationExecutor.RegisterPlugin failed", "plugin", pluginToRegister)
```

File: pkg/kubelet/pluginmanager/reconciler/reconciler.go [hunk 12]
```diff
@@ -122,42 +122,42 @@ func (rc *reconciler) reconcile() {
...
 			}
 			if err == nil {
-				klog.V(1).Infof(pluginToRegister.GenerateMsgDetailed("operationExecutor.RegisterPlugin started", ""))
+				klog.V(1).InfoS(pluginToRegister.GenerateMsgDetailed("operationExecutor.RegisterPlugin started", ""))
```
Comment (serathius) [inline pkg/kubelet/pluginmanager/reconciler/reconciler.go:160]:
```suggestion
				klog.V(1).InfoS("operationExecutor.RegisterPlugin started", "plugin", pluginToRegister)
```

**Comment by qingwave** [comment]
@serathius Fixed as suggested above.

**Pull request merged**